From d27850d688c0b233b9523edfcc34f2b63b40dcd8 Mon Sep 17 00:00:00 2001
From: Xiaoning Wang <xiaoning.wang@nxp.com>
Date: Fri, 24 Aug 2018 09:04:29 +0800
Subject: [PATCH 4461/5242] MLK-19166-5 arm64: dts: imx8qm: delete flexcan
 node for domus

commit  573e09ce982e4b69e3e244a08cc2c07bc3d5fd9b from
https://source.codeaurora.org/external/imx/linux-imx.git

For add flexcan node in dtsi and xen doesn't support can, add "delete
flexcan node" for DomUs now.
Change interrupt-parent to gic, for xen does not support "wu"
interrupt controller.

Signed-off-by: Xiaoning Wang <xiaoning.wang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../dts/freescale/fsl-imx8qm-lpddr4-arm2-domu.dts  |    4 ++++
 .../boot/dts/freescale/fsl-imx8qm-mek-domu.dts     |    4 ++++
 arch/arm64/boot/dts/freescale/fsl-imx8qm-xen.dtsi  |   12 ++++++++++++
 3 files changed, 20 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2-domu.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2-domu.dts
index dba80e3..ab687ff1 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2-domu.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2-domu.dts
@@ -361,6 +361,10 @@
 	status = "okay";
 };
 
+/delete-node/ &flexcan1;
+/delete-node/ &flexcan2;
+/delete-node/ &flexcan3;
+
 &i2c1_lvds1 {
 	reg = <0x1 0x57247000 0x0 0x1000>;
 	status = "okay";
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-domu.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-domu.dts
index 973f875..8bd72c6 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-domu.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-domu.dts
@@ -361,6 +361,10 @@
 	status = "okay";
 };
 
+/delete-node/ &flexcan1;
+/delete-node/ &flexcan2;
+/delete-node/ &flexcan3;
+
 &i2c1_lvds1 {
 	reg = <0x1 0x57247000 0x0 0x1000>;
 	status = "okay";
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-xen.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qm-xen.dtsi
index b4da96a..984e40c 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-xen.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-xen.dtsi
@@ -122,6 +122,18 @@
 	interrupt-parent = <&gic>;
 };
 
+&flexcan1 {
+	interrupt-parent = <&gic>;
+};
+
+&flexcan2 {
+	interrupt-parent = <&gic>;
+};
+
+&flexcan3 {
+	interrupt-parent = <&gic>;
+};
+
 &lpuart0 {
 	interrupt-parent = <&gic>;
 };
-- 
1.7.9.5

